src/components/ClientDialog.tsx

```diff
@@ -0,0 +1,207 @@
+/**
+* ClientDialog.js
+*
+* @2023 Digital Aid Seattle
+*/
+
+import { Button } from "primereact/button";
+import { Dialog } from "primereact/dialog";
+import { InputText } from "primereact/inputtext";
+import { RadioButton } from 'primereact/radiobutton';
+
+import React, { useEffect, useState, useRef} from "react";
+import { TicketType, clientService } from "../services/ClientService";
+import { AnimalType, ClientType, RequestType } from "../types";
+
+const defaultClient: ClientType = {
+  id: null,
+  first_name: '',
+  last_name: '',
+  email: '',
+  phone: '',
+}
+
+const defaultAnimal: AnimalType = {
+  id: null,
+  name: '',
+  species: '',
+  client_id: null,
+}
+
+const defaultRequest: RequestType = {
+  id: null,
+  client_id: null,
+  animal_id: null,
+  service_category: '',
+  source: '',
+  staff_id: null,
+}
+
+const ClientDialog = (props) => {
```

Comment:
(Context: I'm on my first read-through of this code)

I suggest we define an Interface or Type for props. Would make the intended usage and behaviors of the component more clear.

Reply:
Agreed - the repo is in a js - ts transition so there are probably many examples of this.